src/Stratis.Bitcoin.Features.Miner/BlockDefinition.cs

```diff
@@ -170,7 +170,7 @@ private void Configure()
             this.BlockSize = 1000;
             this.BlockTemplate = new BlockTemplate(this.Network);
             this.BlockTx = 0;
-            this.BlockWeight = 4000;
+            this.BlockWeight = 1000 * this.Network.Consensus.Options.WitnessScaleFactor;
```

Comment:
This is not a consensus rule so, I guess this is ok (for now)

Reply:
The only bad consequence that could be here is that we will have less space in a block for txes than we should or we fit too much and the block is not valid. 

As far as I understand initializing BlockWeight with this value is needed just to reserve some room for protocol txes. As long as protocol txes are lighter than 1000 bytes we are cool.

Reply:
Yep @noescape00 nailed it!

Reply:
Ah some witness information is kept in coinbase thats why the increase to 4k?

Reply:
Actually this should not be increased at all. If we are using block weight, it should always be 4000 here regardless of the network

Reply:
I just checked that we are using these in quite messy way. We are using block weight with scaling factor of one which does not make sense

Reply:
But I think the SC guys have very big transactions so the weight limits the number of trx they can push in one block.

Reply:
Perhaps it will be more correct to rename `WitnessScaleFactor`  to `BlockScaleFactor`?